core/src/main/scala/persistentscheduler/PersistentScheduler.scala

```diff
@@ -78,29 +82,30 @@ class PersistentScheduler(persistence: SchedulerPersistence) extends Actor
     self ! State(subscriptions)
   }
 
-  def schedulePersistenceCheckForEvents(): Unit = {
+  def schedulePersistenceCheckForEvents(): Cancellable = {
```

Comment:
can return Unit I guess, because the result is not used anywhere, also check the methods below. I think Unit should be fine everywhere